test/e2e/util_test.go

```diff
@@ -632,25 +632,26 @@ func verifyExternalIngressController(t *testing.T, name types.NamespacedName, ho
 		}
 	}()
 
-	// If we have a DNS as an external IP address, make sure we can resolve it before moving on.
-	// This just limits the number of "could not resolve host" errors which can be confusing.
+	// Wait and check that the dns name resolves first. Takes a long time, so
+	// if the hostname is actually an IP address, skip this.
 	if net.ParseIP(address) == nil {
-		if err := wait.PollImmediate(10*time.Second, 5*time.Minute, func() (bool, error) {
-			_, err := net.LookupIP(address)
+		if err := wait.PollUntilContextTimeout(context.Background(), 10*time.Second, 20*time.Minute, false, func(context context.Context) (bool, error) {
+			_, err := net.LookupHost(address)
 			if err != nil {
-				t.Logf("waiting for loadbalancer domain %s to resolve...", address)
+				t.Logf("%v waiting for HTTP route name %s to resolve (%v)", time.Now(), address, err)
 				return false, nil
 			}
 			return true, nil
 		}); err != nil {
-			t.Fatalf("loadbalancer domain %s was unable to resolve:", address)
+			t.Fatalf("HTTP route name %s was unable to be resolved: %v", address, err)
 		}
 	}
```

Comment:
`assertHttpRouteConnection` has similar logic.  Can you factor this logic out into a separate `waitForHostToResolve` helper, and use that helper in both `assertHttpRouteConnection` and `verifyExternalIngressController`?